### DISCUSSION TITLE
Treatlife Devices

### DISCUSSION THREAD

**Discussion by DocDrydenn** [make-all/tuya-local#1217] [Ideas]
[Treatlife](https://www.treatlife.tech/)

Okay, so for whatever reason (probably overall cost?), a few years ago, I went "all in" on the Treatlife products for my home. After finally "seeing the light" that was Home Assistant, I've jumped in with all my devices - somewhere around 70 devices. (also about 15 other non-Treatlife Tuya devices)

Specifically: (product_name)
- 18x Smart Light Switch SS02S
- 28x Smart Mini Plug_SK50
- 3x DS02S Smart dimmer switch
- 1x Multicolor Ambiance Table Lamp-FL20
- 5x Ceiling Fan Dimmer Switch DS03
- 1x Smart Dimmer Plug DP20
- 5x LED Strip Lights
- 4x SS02S smart light switch
- 2x SS02 3-Way Smart Wi-Fi Light Switch
- 2x Smart Dimmer Plug with 2 Sockets

Anyway, everything seems to be working fine, however, I'm still getting errors in the Home Assistant logs. Maybe I'm doing something wrong in the yaml or reading the Tuya data incorrectly?...

I'd like to submit my data/yamls for *review* and possible inclusion into future updates. If that's cool, I'll submit each device to this discussion thread in its own comment. Each will include all of the required Tuya info and the yaml I have attempted to create.

Like I said, everything seems to be working, so I didn't see the need to open an issue for them.

**Comment by make-all** [comment]
How many of these need new configs?  Lights and switches are most often following the same generic dps layout as other similar devices.  I don't want to add new configs just because the brand is different, as there are already close to 500 configs, which is far more than I ever envisaged supporting when I started this.
